src/app/navbar/navbar.component.ts

```diff
@@ -9,10 +10,22 @@ import { LanguageService } from '../language.service';
 })
 export class NavbarComponent {
 
+  treeHighlighted: "highlighted" | "";
+  cardHighlighted: "highlighted" | "";
   constructor(
     public readonly authService: AuthenticationService,
-    private readonly langService: LanguageService
-  ) { }
+    private readonly langService: LanguageService,
+    private router: Router
+  ) {
+    const treetestUrls = ['/create-tree-test', '/tests'];
+    const treetestPartialUrls = ['/create-tree-test/', '/treetest/', '/results/', '/pie-tree/'];
+    router.events.subscribe((val) => {
+      if (!(val instanceof NavigationEnd)) return;
+      const partialMatch = treetestPartialUrls.find(url => val.urlAfterRedirects.startsWith(url));
+      this.treeHighlighted = treetestUrls.includes(val.urlAfterRedirects) || partialMatch ? "highlighted" : ""
+      this.cardHighlighted = this.treeHighlighted || val.urlAfterRedirects === "/admin" ? "" : "highlighted";
+    });
```

Comment:
I think a better solution for this would be to use routerLinkActive (https://angular.io/api/router/RouterLinkActive).

Reply:
Thanks for the hint, I did not know this option before. If I am not mistaken this would mean we should 
rearrange the url structure? Like pack all treetest things in /treetest/... and all cardsort urls in /cardsort/.. ?
I would also find this url structure much cleaner, just wanted to check it it is ok to change that?

Reply:
Yes, you are right. This would imply changing the url structure to nested routes. I would also prefer this url structure, but I need to check if this would have any side-effects. I suggest, that we implement this in a different branch and stick with your current solution for now.